spec/unit/type/service_spec.rb

```diff
@@ -67,6 +67,12 @@ def safely_load_service_type
     expect(svc.should(:ensure)).to eq(:stopped)
   end
 
+  describe 'the absent property' do
+    it 'should fail validate if it is a service' do
+      expect { Puppet::Type.type(:service).new(:name => "service_name", :absent => 'sure') }.to raise_error(Puppet::Error)
```

Comment:
I think you want to try setting `:ensure => :absent` and then adding the expected exception message, e.g. `raise_error(Puppet::Error, /Managing absent on a service is not supported/)` Otherwise the code could be raising a `Puppet::Error` for an unrelated reason and the test will still pass.

Reply:
Not sure if my previous comment made sense? The code:

```ruby
Puppet::Type.type(:service).new(:name => "service_name", :absent => 'sure')
```
does raise a `Puppet::Error`, but it's not for the reason that we're trying to test. If you remove the `expect { }` wrapper, then the raised exception will have the following message:

```
     Puppet::Error:
       no parameter named 'absent'
```

To accurately test that we can't do `ensure => absent` in a manifest, then you'll want to change the test to:

```suggestion
      expect { Puppet::Type.type(:service).new(:name => "service_name", :ensure => :absent) }.to raise_error(Puppet::Error, /Managing absent on a service is not supported/)
```

That way we're sure the test is exercising the code we expect it to (and not failing due to some other `Puppet::Error`)